synapse/events/third_party_rules.py

```diff
@@ -370,3 +380,37 @@ async def _get_state_map_for_room(self, room_id: str) -> StateMap[EventBase]:
             state_events[key] = room_state_events[event_id]
 
         return state_events
+
+    async def on_profile_update(
+        self, user_id: str, new_profile: ProfileInfo, by_admin: bool
+    ) -> None:
+        """Called after the global profile of a user has been updated. Does not include
+        per-room profile changes.
+
+        Args:
+            user_id: The user whose profile was changed.
+            new_profile: The updated profile for the user.
+            by_admin: Whether the profile update was performed by a server admin.
+        """
+        for callback in self._on_profile_update_callbacks:
+            try:
+                await callback(user_id, new_profile, by_admin)
+            except Exception as e:
+                logger.exception(
+                    "Failed to run module API callback %s: %s", callback, e
+                )
+
+    async def on_deactivation(self, user_id: str, by_admin: bool) -> None:
+        """Called after a user has been deactivated.
+
+        Args:
+            user_id: The deactivated user.
+            by_admin: Whether the deactivation is performed by a server admin.
```

Comment:
```suggestion
            by_admin: Whether the deactivation was performed by a server admin.
```

as the rest of the docstring is in past tense.